### PR TITLE
Refactor cddl specification

### PR DESCRIPTION
**Pull request by coot**
* per mini-protocol CDDL spec
* tx-submission2 CDDL spec
* keep-alive CDDL spec
* reimplement handshake spec: node-to-node & node-to-client
* update technical repor:
  - fixed a missplaced section
  - include CDDL mini-protocol encoding specs
  - standardise message and state indentifiers
  - link to haddock documentation

### CODE REVIEW

File: docs/network-spec/miniprotocols.tex [hunk 1]
```diff
@@ -241,15 +241,20 @@ \section{Overview of all implemented Mini Protocols}
     {ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Type.hs}
     {https://input-output-hk.github.io/ouroboros-network/ouroboros-network/Ouroboros-Network-Protocol-LocalStateQuery-Type.html\#t:LocalStateQuery}
 
+\section{CBOR and CDDL}
+All mini-protocols encoded using coincise binary object representation (CBOR),
```
Comment (MarcFontaine) [inline docs/network-spec/miniprotocols.tex:245]: All mini-protocol messages are encoded.. ?

Comment (coot) [reply docs/network-spec/miniprotocols.tex:245]: yes

**Review by MarcFontaine — APPROVED** [review]
Cool to see that theses parts of the  code are still maintained.
LGTM

**Review by njd42 — APPROVED** [review]
LGTM - nice refactor that brings things up to date

**Comment by coot** [comment]
bors merge

**Pull request merged**